src/bindings/c/ov/include/c_api/ov_c_api.h

```diff
@@ -0,0 +1,1122 @@
+
+/**
+ * @brief Add resize operation to model's dimensions. 
+ * @param preprocess_input_process_steps A pointer to ov_preprocess_input_process_steps_t.
+ * @param resize_algorithm A ov_preprocess_resizeAlgorithm instance
+ * @return Status code of the operation: OK(0) for success.
+ */
+OPENVINO_C_API(ov_status_e) ov_preprocess_input_resize(ov_preprocess_input_process_steps_t* preprocess_input_process_steps,
+                                                        const ov_preprocess_resize_algorithm_e resize_algorithm);
+
+/**
+ * @brief Set ov_preprocess_input_tensor_info_t precesion. 
+ * @param preprocess_input_tensor_info A pointer to the ov_preprocess_input_tensor_info_t.
+ * @param element_type A point to element_type
+ */
+OPENVINO_C_API(ov_status_e) ov_preprocess_input_tensor_info_set_element_type(ov_preprocess_input_tensor_info_t* preprocess_input_tensor_info,
+                                                        const ov_element_type_e element_type);
+
+/**
+ * @brief Helper function to reuse element type and shape from user's created tensor. 
+ * @param preprocess_input_tensor_info A pointer to the ov_preprocess_input_tensor_info_t.
+ * @param tensor A point to ov_tensor_t
+ */
+OPENVINO_C_API(ov_status_e) ov_preprocess_input_tensor_info_set_tensor(ov_preprocess_input_tensor_info_t* preprocess_input_tensor_info,
+                                                        const ov_tensor_t* tensor);
+
+/**
+ * @brief Set ov_preprocess_input_tensor_info_t layout. 
+ * @param preprocess_input_tensor_info A pointer to the ov_preprocess_input_tensor_info_t.
+ * @param layout A point to ov_layout_t
+ */
+OPENVINO_C_API(ov_status_e) ov_preprocess_input_tensor_info_set_layout(ov_preprocess_input_tensor_info_t* preprocess_input_tensor_info,
+                                                        const ov_layout_t layout);
+
+/**
+ * @brief Get the output info of ov_preprocess_output_info_t instance. 
+ * @param preprocess A pointer to the ov_preprocess_t.
+ * @param preprocess_output_info A pointer to the ov_preprocess_output_info_t.
+ * @return Status code of the operation: OK(0) for success.
+ */
+OPENVINO_C_API(ov_status_e) ov_preprocess_get_output_info(const ov_preprocess_t* preprocess,
+                                                ov_preprocess_output_info_t **preprocess_output_info);
+
+/**
+ * @brief Get the output info of ov_preprocess_output_info_t instance. 
+ * @param preprocess A pointer to the ov_preprocess_t.
+ * @param tensor_index The tensor index
+ * @param preprocess_output_info A pointer to the ov_preprocess_output_info_t.
+ * @return Status code of the operation: OK(0) for success.
+ */
+OPENVINO_C_API(ov_status_e) ov_preprocess_get_output_info_by_index(const ov_preprocess_t* preprocess,
+                                                const size_t tensor_index,
+                                                ov_preprocess_output_info_t **preprocess_output_info);
```

Comment:
maybe we can skip preprocessing in C-API at all? do we have customers for that?
I think we can recommend customers to add preprocessing steps in python API, then save the model. After such model can be loaded with cache_dir in C API (see https://docs.openvino.ai/latest/openvino_docs_OV_UG_Preprocess_Usecase_save.html)